File: c#/time-deposit-kata-net/Domain/Entities/TimeDeposit.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace time_deposit_kata_net
{
    public class TimeDeposit
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string PlanType { get; set; } = string.Empty;

        [Required]
        public double Balance { get; set; }

        [Required]
        public int Days { get; set; }

        public List<Withdrawal> Withdrawals { get; set; } = new List<Withdrawal>();
    }
}

File: java/src/main/java/org/ikigaidigital/TimeDeposit.java
package org.ikigaidigital;

import java.util.ArrayList;
import java.util.List;
import org.ikigaidigital.domain.model.Withdrawal;

public class TimeDeposit {
    private int id;
    private String planType;
    private Double balance;
    private int days;
    private List<Withdrawal> withdrawals = new ArrayList<>();

    public TimeDeposit(int id, String planType, Double balance, int days) {
        this.id = id;
        this.planType = planType;
        this.balance = balance;
        this.days = days;
    }

    public int getId() { return id; }

    public String getPlanType() {
        return planType;
    }

    public Double getBalance() {
        return balance;
    }

    public int getDays() {
        return days;
    }

    public void setBalance(Double balance) {
        this.balance = balance;
    }

    public List<Withdrawal> getWithdrawals() {
        return withdrawals;
    }

    public void setWithdrawals(List<Withdrawal> withdrawals) {
        this.withdrawals = withdrawals;
    }

}
